csv/src/main/java/com/fasterxml/jackson/dataformat/csv/impl/UTF8Reader.java

```diff
@@ -403,19 +403,6 @@ private boolean loadMore(int available) throws IOException
         // Bytes that need to be moved to the beginning of buffer?
         if (available > 0) {
             if (_inputPtr > 0) {
-                if (!canModifyBuffer()) {
```

Comment:
Why do you feel this is no longer needed? This is to prevent reader from modifying buffer caller gave, in case moving would occur.

If anything, looks like it should avoid copying even if `_inputSource` was not null.

Reply:
The test pass without this check. The existing tests and we gain an extra working test (the 497 test).

I don't know how to rewrite this because I don't have a broken test to work with. My main aim is not to have the 497 test (which is a valid test) fail. I'm not against revisiting this but since this change breaks no tests (other than minor woring diff in exception message on 1 test) and fixes one broken test, I'd prefer to remove this check for now. If this was merged, I would follow up and remove the same check in the TOML and YAML code bases because there is no proof that this check helps there either.

Reply:
It'd be nice to have a test to verify underlying buffer is not modified when it shouldn't be -- which is what check does -- but even without that we should not make change that will now unexpectedly do that.

Specifically this is the case where caller passes `byte[]` containing UTF-8 content: expectation is that this buffer is not modified. And as such it should not be.

Reply:
* ok, so I extended the tests to see if array input gets modified and it does
* is there anyway that the parser could be modified to clone the array when we hit this case and to work with the cloned array - to avoid modifying the input array?

Reply:
I added back the code check for whether the buffer is modifiable - but instead of throwing an exception, I clone the buffer to avoid modifying the original. The cloning only happens in rare cases but I think it is worth the runtime hit to fix the broken test.

Reply:
I went ahead and used the idea, just rewrote slightly. Thank you for test to verify buffer not modified!

Ideally wouldn't need to move things around, will see what I can do there. But at least I can go ahead and merge this now.

Reply:
Actually, can just remove copying action altogether.